frontend/src/pages/modelServing/customServingRuntimes/CustomServingRuntimeListView.tsx

```diff
@@ -1,59 +1,65 @@
 import * as React from 'react';
 import { useNavigate } from 'react-router';
 import { Button, ToolbarItem } from '@patternfly/react-core';
-import Table from '~/components/Table';
+import styles from '@patternfly/react-styles/css/components/Table/table';
 import { TemplateKind } from '~/k8sTypes';
 import { patchDashboardConfigTemplateOrder } from '~/api';
 import { useDashboardNamespace } from '~/redux/selectors';
 import useNotification from '~/utilities/useNotification';
-import { getDragItemOrder, getServingRuntimeNameFromTemplate } from './utils';
+import Table from '~/components/Table';
+import useDraggableTable from '~/utilities/useDraggableTable';
+import { getServingRuntimeNameFromTemplate, getSortedTemplates } from './utils';
 import DeleteCustomServingRuntimeModal from './DeleteCustomServingRuntimeModal';
 import { columns } from './templatedData';
 import CustomServingRuntimeTableRow from './CustomServingRuntimeTableRow';
 import { CustomServingRuntimeContext } from './CustomServingRuntimeContext';
 
 const CustomServingRuntimeListView: React.FC = () => {
   const {
-    servingRuntimeTemplateOrder: { data: templateOrder },
-    servingRuntimeTemplates: { data: templates },
+    servingRuntimeTemplateOrder: { data: templateOrder, refresh: refreshOrder },
+    servingRuntimeTemplates: { data: unsortedTemplates },
     refreshData,
   } = React.useContext(CustomServingRuntimeContext);
   const { dashboardNamespace } = useDashboardNamespace();
   const notification = useNotification();
   const navigate = useNavigate();
+  const bodyRef = React.useRef<HTMLTableSectionElement>(null);
   const [deleteTemplate, setDeleteTemplate] = React.useState<TemplateKind>();
-  const [dragItemOrder, setDragItemOrder] = React.useState(
-    getDragItemOrder(templates, templateOrder),
+  const sortedTemplates = React.useMemo(
+    () => getSortedTemplates(unsortedTemplates, templateOrder),
+    [unsortedTemplates, templateOrder],
+  );
+  const setItemOrder = React.useCallback(
+    (itemOrder: string[]) => {
+      patchDashboardConfigTemplateOrder(itemOrder, dashboardNamespace)
+        .then(refreshOrder)
+        .catch((e) => notification.error(`Error update the serving runtimes order`, e.message));
+    },
+    [dashboardNamespace, refreshOrder, notification],
   );
 
-  React.useEffect(() => {
-    patchDashboardConfigTemplateOrder(dragItemOrder, dashboardNamespace).catch((e) =>
-      notification.error(`Error update the serving runtimes order`, e.message),
-    );
-  }, [dragItemOrder, dashboardNamespace, notification]);
-
-  React.useEffect(() => {
-    setDragItemOrder(getDragItemOrder(templates, templateOrder));
-  }, [templates, templateOrder]);
+  const { isDragging, tbodyDragFunctions, trDragFunctions } = useDraggableTable(
```

Comment:
Recommend that you look to see how you can put everything inside the `useDraggableTable`... your hook should set you up for _everything_ you need. The output of a hook could be a series of `row` and `table` props -- any use of those props could be entirely managed within' the hook itself.

Highly recommend you take a look at your component here and understand the setup you're doing here in the `CustomServingRuntimeListView` for this hook -- can not all of this be within it?